manager/controllers/app/select_modules.go

```diff
@@ -50,9 +50,16 @@ func (p *PathBuilder) solve() (Solution, error) {
 // Then, transformations are added to the found paths, and clusters are matched to satisfy restrictions from admin config policies.
 // Optimization is done by the shortest path (the paths are sorted by the length). To be changed in future versions.
 func (p *PathBuilder) FindPaths() []Solution {
+	var protocol taxonomy.ConnectionType
+	// If the connection name is empty, the default protocol is s3.
+	if p.Asset.DataDetails.Details.Connection.Name == "" {
+		protocol = "s3"
+	} else {
+		protocol = p.Asset.DataDetails.Details.Connection.Name
+	}
 	NodeFromAssetMetadata := Node{
 		Connection: &taxonomy.Interface{
-			Protocol:   p.Asset.DataDetails.Details.Connection.Name,
+			Protocol:   protocol,
 			DataFormat: p.Asset.DataDetails.Details.DataFormat,
```

Comment:
Where is this coming from? Changes to format are done after the path has been found.

Reply:
@shlomitk1 please explain, in this line we changed protocol not dataformat.

Reply:
p.Asset.DataDetails.Details.DataFormat is not set. If you want to use an empty string, do it explicitly.

Reply:
@shlomitk1 this line already exists in the current code: https://github.com/fybrik/fybrik/blob/master/manager/controllers/app/select_modules.go#L56
it does not refer only to the case where dataformat is empty.

Reply:
What I mean is that you change protocol, but not data format. Both are not set. It's better to set values explicitly and not rely on an empty field value.

Reply:
Fixed. Thanks